src/Microsoft.Data/DataFrame.cs

```diff
@@ -301,116 +352,62 @@ public DataFrame Sort(string columnName, bool ascending = true)
-                }
+                BaseColumn column = Column(i);
+                _table.SetColumnName(column, prefix + column.Name);
             }
-            else if (joinAlgorithm == JoinAlgorithm.FullOuter)
+            return this;
+        }
+
+        /// <summary>
+        /// Adds a suffix to the column names
+        /// </summary>
+        public DataFrame AddSuffix(string suffix)
+        {
+            for (int i = 0; i < ColumnCount; i++)
             {
-                long newRowCount = Math.Max(RowCount, other.RowCount);
-                long numberOfNulls = newRowCount - RowCount;
-                for (int i = 0; i < ColumnCount; i++)
-                {
-                    BaseColumn newColumn = Column(i).Clone(numberOfNullsToAppend: numberOfNulls);
-                    ret.InsertColumn(ret.ColumnCount, newColumn);
-                }
-                numberOfNulls = newRowCount - other.RowCount;
-                for (int i = 0; i < other.ColumnCount; i++)
-                {
-                    BaseColumn newColumn = other.Column(i).Clone(numberOfNullsToAppend: numberOfNulls);
-                    SetSuffixForDuplicatedColumnNames(ret, newColumn, leftSuffix, rightSuffix);
-                    ret.InsertColumn(ret.ColumnCount, newColumn);
-                }
+                BaseColumn column = Column(i);
+                _table.SetColumnName(column, column.Name + suffix);
             }
-            else if (joinAlgorithm == JoinAlgorithm.Inner)
+            return this;
+        }
+
+        /// <summary>
+        /// Returns a random sample of rows
+        /// </summary>
+        /// <param name="numberOfRows">Number of rows in the returned DataFrame</param>
+        public DataFrame Sample(int numberOfRows)
+        {
+            Random rand = new Random();
+            PrimitiveColumn<long> indices = new PrimitiveColumn<long>("Indices", numberOfRows);
+            for (long i = 0; i < numberOfRows; i++)
             {
-                long newRowCount = Math.Min(RowCount, other.RowCount);
-                PrimitiveColumn<long> mapIndices = new PrimitiveColumn<long>("mapIndices", newRowCount);
-                for (long i = 0; i < newRowCount; i++)
-                {
-                    mapIndices[i] = i;
-                }
-                for (int i = 0; i < ColumnCount; i++)
-                {
-                    BaseColumn newColumn = Column(i).Clone(mapIndices);
-                    ret.InsertColumn(ret.ColumnCount, newColumn);
-                }
-                for (int i = 0; i < other.ColumnCount; i++)
-                {
-                    BaseColumn newColumn = other.Column(i).Clone(mapIndices);
-                    SetSuffixForDuplicatedColumnNames(ret, newColumn, leftSuffix, rightSuffix);
-                    ret.InsertColumn(ret.ColumnCount, newColumn);
-                }
+                indices[i] = rand.Next((int)Math.Min(Int32.MaxValue, RowCount));
```

Comment:
You probably don't need to call `(int)Math.Min(Int32.MaxValue, RowCount)` each time through the loop, do you?